### PR TITLE
HSEARCH-3089 Add to the DSL the missing query predicates (excluding spatial, faceting, moreLikeThis) compared to Search 5

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3089

Waiting for HSEARCH-3256 (#1902), I will probably have to make some change to rebase on that PR.

Also requires #1904 .

### CODE REVIEW

**Comment by yrodiere** [comment]
@fax4ever Rebased and ready for review!

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/analysis/impl/ScopedAnalyzer.java [hunk 1]
```diff
@@ -10,21 +10,22 @@
...
+import org.apache.lucene.analysis.DelegatingAnalyzerWrapper;
+import org.apache.lucene.analysis.core.KeywordAnalyzer;
+
 import org.hibernate.search.util.common.impl.CollectionHelper;
 
 /**
  * @author Guillaume Smet
  */
-public final class ScopedAnalyzer extends AnalyzerWrapper {
+public final class ScopedAnalyzer extends DelegatingAnalyzerWrapper {
```
Comment (fax4ever) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/analysis/impl/ScopedAnalyzer.java]: Very good simplification here. There's always time to make things more complex ;)

File: engine/src/main/java/org/hibernate/search/engine/reporting/impl/EngineEventContextMessages.java [hunk 2]
```diff
@@ -62,6 +63,9 @@
 	@Message(value = "field '%1$s'")
 	String indexFieldAbsolutePath(String absolutePath);
 
+	@Message(value = "fields %1$s")
+	String indexFieldAbsolutePaths(List<String> absolutePaths);
```
Comment (fax4ever) [inline engine/src/main/java/org/hibernate/search/engine/reporting/impl/EngineEventContextMessages.java]: I like this change. It improves consistency.

File: engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/MatchPredicateContext.java [hunk 3]
```diff
@@ -12,7 +12,51 @@
...
+	 * Fuzziness allows to match documents that do not contain the value to match,
+	 * but a close value, for example with one letter that differs.
+	 *
+	 * @param maxEditDistance The maximum value of the edit distance, which defines how permissive the fuzzy predicate will be.
+	 * <p>
+	 * Roughly speaking, the edit distance is the number of changes between two terms: switching characters, removing them, ...
+	 * <p>
+	 * If zero, then fuzziness is completely disabled.
+	 * The other accepted values, {@code 1} and {@code 2}, are increasingly fuzzy.
+	 * @param nonFuzzyPrefixLength Length of the prefix for which no fuzziness will be applied.
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/MatchPredicateContext.java]: You're right, this is better. Will do.

File: integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/MatchSearchPredicateIT.java [hunk 4]
```diff
@@ -468,7 +470,226 @@ public void predicateLevelBoost_multiFields() {
 	}
 
 	@Test
-	public void multi_fields() {
+	@PortedFromSearch5(original = "org.hibernate.search.test.dsl.DSLTest.testFuzzyQuery")
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/MatchSearchPredicateIT.java]: You're right, will do.

File: engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/SimpleQueryStringPredicateFieldSetContext.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ */
+package org.hibernate.search.engine.search.dsl.predicate;
+
+/**
+ * The context used when defining a simple query string predicate, after at least one field was mentioned.
+ */
+public interface SimpleQueryStringPredicateFieldSetContext extends MultiFieldPredicateFieldSetContext<SimpleQueryStringPredicateFieldSetContext> {
+
+	/**
+	 * Define the default operator as AND.
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/SimpleQueryStringPredicateFieldSetContext.java]: I will mention the OR operator explicitly in the explanation just below, and will add a note in the javadoc of `org.hibernate.search.engine.search.dsl.predicate.SearchPredicateFactoryContext#simpleQueryString`.

File: integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/SimpleQueryStringSearchPredicateIT.java [hunk 6]
```diff
@@ -0,0 +1,799 @@
...
+		initData();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-2678")
+	@PortedFromSearch5(original = "org.hibernate.search.test.dsl.SimpleQueryStringDSLTest.testSimpleQueryString")
+	public void simpleQueryString() {
+		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
+		String absoluteFieldPath = indexMapping.analyzedStringField1.relativeFieldName;
+		Function<String, SearchQuery<DocumentReference>> createQuery = queryString -> searchTarget.query().asReference()
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/SimpleQueryStringSearchPredicateIT.java]: I agree it's not idea, but copy/pasting the query multiple times in the same methods, only changing the query string, makes things worse in my opinion. I'll keep it that way for now.

**Comment by yrodiere** [comment]
@fax4ever Thanks for the review, good points. I amended some commits to follow your suggestions and I force-pushed.

**Comment by fax4ever** [comment]
Merged thanks!

**Pull request merged**